### PR TITLE
fix #833 field array dirty formState

### LINKED ISSUE #833: Calling useFieldArray hook methods does not update formState.dirty value
**Describe the bug**

Calling useFieldArray hook methods does not update formState.dirty value.

**To Reproduce**
1. Go to https://codesandbox.io/s/react-hook-form-usefieldarray-popep
2. Click on APPEND
3. Observe that formState hasn't changed

**Codesandbox link**
https://codesandbox.io/s/react-hook-form-usefieldarray-popep

**Expected behavior**
append, prepend, remove, swap, move, insert should all update dirty state to true

**Desktop (please complete the following information):**
 - OS: Windows
 - Browser Chrome Version 79.0.3945.117 (Official Build) (64-bit)
 - Version 4.5.2

**Additional context**

### PR DESCRIPTION
**Pull request by bluebill1049**

### CODE REVIEW

**Comment by bluebill1049** [comment]
https://codesandbox.io/s/react-hook-form-usefieldarray-yikiz

pretty stable now :)

**Comment by bluebill1049** [comment]
@JeromeDeLeon @kotarella1110 please take a look at this PR if you guys have sometimes. We can get this boy out this weekend.

**Comment by JeromeDeLeon** [comment]
When I moved the `console.log` in between `useForm` and `useFieldArray`, `dirty` is working and set to `true` when I appended a row; however, when the `log` is below the custom hooks, `dirty` was not setting correctly on first append scenario and will only be set to `true` after the second one.

**Comment by bluebill1049** [comment]
> When I moved the `console.log` in between `useForm` and `useFieldArray`, `dirty` is working and set to `true` when I appended a row; however, when the `log` is below the custom hooks, `dirty` was not setting correctly on first append scenario and will only be set to `true` after the second one.

thanks @JeromeDeLeon you have a screenshot of the actual error.

**Comment by JeromeDeLeon** [comment]
Not Working
image

Working
image

**Comment by bluebill1049** [comment]
hmm looks like it's broken now, i must have broke it.

**Comment by bluebill1049** [comment]
should be better now: https://github.com/react-hook-form/react-hook-form/pull/855

still need to fix onBlur

File: src/useFieldArray.ts [hunk 1]
```diff
@@ -28,13 +33,16 @@ export function useFieldArray<
   const [fields, setField] = React.useState<
     WithFieldId<Partial<FormArrayValues>>[]
   >(mapIds(memoizedDefaultValues));
+  const shouldCheckDirty = readFormStateRef.current.dirty;
```
Comment (JeromeDeLeon) [inline src/useFieldArray.ts]: Instead of doing this, why not read it real-time?

```js
if (readFormStateRef.current.dirty) {
  // logic
}
```

**Review by JeromeDeLeon — CHANGES_REQUESTED** [review]

File: src/useFieldArray.ts (hunk 1 above)
Comment (bluebill1049) [reply src/useFieldArray.ts]: yea i am fixing it. you r right.

**Comment by JeromeDeLeon** [comment]
Yes, it's working now. What was the cause of it? @bluebill1049

**Comment by bluebill1049** [comment]
> Yes, it's working now. What was the cause of it? @bluebill1049

your comment above :)

File: src/useForm.ts [hunk 2]
```diff
@@ -223,7 +223,7 @@ export function useForm<FormValues extends FieldValues = FieldValues>({
   );
 
   const setDirty = (name: FieldName<FormValues>): boolean => {
```
Comment (JeromeDeLeon) [inline src/useForm.ts]: Are we fine doing this? I mean, I just thought we're just setting a value here therefore, do not expect to return anything here. Better if we could rename to something like `verifyDirty` or something like that.

Comment (bluebill1049) [reply src/useForm.ts]: you are right. the return is for deciding re-render or not. feel free to come up with a better name for it.

**Pull request merged**